### PR TITLE
Feature/brentonmdunn/get viz by title

### LINKED ISSUE #31: Make Viz title part of URL for individual viz page
Title should be in url and then extracted in component to make backend call.

### PR DESCRIPTION
**Pull request by brentonmdunn**
### Tracking Info

Resolves #31 

Make sure your branch name conforms to: `<feature|staging|bugfix|...>/<username>/<3-4 word description separated by dashes>`. Otherwise, please rename your branch and create a new PR.

### Changes

What changes did you make?

* Added a `title=` query field the URL which is used to call the backend
* Created a 404 page which is rendered if the URL and/or the query is wrong or incomplete

### Testing

How did you confirm your changes worked?

* Tested with various URLs to confirm that error handling was correct
* Verified functionality with sample visualization currently in database

### Confirmation of Change

Upload a screenshot, if possible. Otherwise, please provide instructions on how to see the change.

workingURL

404URL

### CODE REVIEW

File: SDCTA/src/api/consumer.ts [hunk 1]
```diff
@@ -1,3 +1,24 @@
+import { type VisualizationObject } from "./data";
+export const API_URL = process.env.REACT_APP_API_URL;
+export const GET_VIS = `${API_URL ?? "http://localhost:3001"}/visualization`;
+export const getVisByTitle = async (
+  title: string
+): Promise<VisualizationObject | null> => {
```
Comment (NirmalAgnihotri) [inline SDCTA/src/api/consumer.ts:6]: Great job on this return type

File: SDCTA/src/pages/IndividualVisualization.tsx [hunk 2]
```diff
@@ -1,63 +1,75 @@
...
+  const [title] = useState(searchParams.get("title"));
+  const [visObj, setVisObj] = useState<VisualizationObject | null>(
+    initialState
+  );
 
-interface IndividualVisualizationProp {
-  url: string;
-  csvlink: string;
-}
+  if (title === "" || title === null) {
```
Comment (NirmalAgnihotri) [inline SDCTA/src/pages/IndividualVisualization.tsx:25]: minor: empty string is falsy in JS so this could be condensed to `if (!title) {...}`

**Review by NirmalAgnihotri — APPROVED** [review]
Overall really well done! Great job successfully making a call to the backend and using the url params.

**Review by aammya8 — APPROVED** [review]
Looks good! Nice that you added the ErrorBoundary class

**Pull request merged**